lib/resources/groups.rb

```diff
@@ -154,6 +158,35 @@ def groups
     end
   end
 
+  # OSX uses opendirectory for groups, so `/etc/group` may not be accurate
+  class DarwinGroup < GroupInfo
+    def groups
+      raw_output = inspec.command('dscacheutil -q group').stdout
+      raw_group_info = raw_output.split("\n\n")
+
+      group_info = []
+      regex = /^([^:]*?)\s*:\s(.*?)\s*$/
+      raw_group_info.each do |data|
+        group_info << inspec.parse_config(data, assignment_regex: regex).params
+      end
+
+      match_etc_group_output(group_info)
+    end
+
+    # Converts the `dscacheutil` groups to match `inspec.etc_group.entries`
+    def match_etc_group_output(dscacheutil_groups)
+      dscacheutil_groups.each { |g| g['gid'] = g['gid'].to_i }
+
+      # Convert `users` to `members` and ` ` to `,`
+      dscacheutil_groups.each do |g|
+        next if g['users'].nil?
+        g['members'] = g['users']
+        dscacheutil_groups.delete(:users)
```

Comment:
This seems curious to me... In the line above, you use the string version of the key, but then you delete the symbol?

Reply:
Shoot, missed that. Great catch. This coupled with the tip you gave me a bout `#delete` cleans this up quite nicely.